game-backend/src/helper/fields.ts

```diff
@@ -0,0 +1,41 @@
+import { State } from '../../../darwin-types/State';
+import Position from '../../../darwin-types/Position';
+import { ARENA_WIDTH, ARENA_HEIGHT } from '../../../darwin-types/Arena';
+
+export const getFreeFields = (state: State): Position[] => {
+  const fields: Position[] = new Array(ARENA_WIDTH)
+    .fill(undefined)
+    .reduce(
+      (acc, _, i) => [
+        ...acc,
+        ...new Array(ARENA_HEIGHT)
+          .fill(undefined)
+          .map((_, j) => ({ x: i, y: j })),
+      ],
+      []
+    );
+  const occupiedFields: Record<string, boolean> = state.objectIds.reduce(
```

Comment:
What about extracting the steps (`getFlatFieldArray`, `getOccupiedFieldMap`) to separate functions? This makes the general logic easier to understand without knowing implementation details.